analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/PermissiveCors.cs

```diff
@@ -0,0 +1,145 @@
+        {
+            var attribute = (AttributeSyntax)context.Node;
+            if (attribute.IsKnownType(KnownType.System_Web_Http_Cors_EnableCorsAttribute, context.SemanticModel)
+                && IsStar(attribute.ArgumentList.Arguments[0].Expression, context.SemanticModel))
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, attribute.GetLocation()));
+            }
+        }
+
+        private static bool IsFirstArgumentAccessControlAllowOrigin(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
+            invocation.ArgumentList.Arguments.First().Expression switch
+            {
+                LiteralExpressionSyntax literal => literal.Token.ValueText == AccessControlAllowOriginHeader,
+                MemberAccessExpressionSyntax memberAccess => IsAccessControlAllowOriginProperty(memberAccess, semanticModel),
+                _ => false
+            };
+
+        private static bool IsAccessControlAllowOriginProperty(MemberAccessExpressionSyntax memberAccess, SemanticModel semanticModel) =>
+            memberAccess.Name.Identifier.Text == AccessControlAllowOriginPropertyName
+            && memberAccess.Expression.IsKnownType(KnownType.Microsoft_Net_Http_Headers_HeaderNames, semanticModel);
+
+        private static bool IsSecondArgumentStarString(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
+            IsStar(invocation.ArgumentList.Arguments[1].Expression, semanticModel);
+
+        private static bool IsStar(ExpressionSyntax expressionSyntax, SemanticModel semanticModel) =>
+            expressionSyntax switch
+            {
+                LiteralExpressionSyntax literal => ContainsStar(semanticModel.GetConstantValue(literal)),
+                IdentifierNameSyntax identifier => ContainsStar(semanticModel.GetConstantValue(identifier)),
+                ImplicitArrayCreationExpressionSyntax arrayCreation => ContainsStar(arrayCreation.Initializer.Expressions, semanticModel),
+                ObjectCreationExpressionSyntax objectCreation => ContainsStar(objectCreation, semanticModel),
+                _ => false
+            };
+
+        private static bool ContainsStar(IEnumerable<ExpressionSyntax> expressions, SemanticModel semanticModel) =>
+            expressions.Any(expression => ContainsStar(semanticModel.GetConstantValue(expression)));
+
+        private static bool ContainsStar(Optional<object> constantValue) =>
+            constantValue is {HasValue: true, Value: StarConstant};
+
+        private static bool ContainsStar(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel) =>
+            (objectCreation.IsKnownType(KnownType.Microsoft_Extensions_Primitives_StringValues, semanticModel)
+             || objectCreation.IsKnownType(KnownType.Microsoft_AspNetCore_Cors_Infrastructure_CorsPolicyBuilder, semanticModel))
+            && objectCreation.ArgumentList is { }
```

Comment:
this can only be null for VB, right?

Reply:
I've checked the code and it is defined as nullable reference also in the `CSharp` namespace:
![image](https://user-images.githubusercontent.com/56015273/115520429-70684400-a28a-11eb-8c6c-85004e289c46.png)
I prefer to let this as it is.

Reply:
do you have a test case for this?

Reply:
I've added one now.